design/one-pager-resource-connectivity-mvp.md

```diff
@@ -638,7 +638,279 @@ parameters:
+  namespace: crossplane-system
+  name: my-gateway
+spec:
+  vpc:
+    referenceType: apiObject
+    value: my-vpc.crossplane-system
+--
+apiVersion: network.aws.crossplane.io/v1alpha1
+kind: AwsRouteTable
+metadata:
+  namespace: crossplane-system
+  name: my-rt
+spec:
+  vpcId:
+    referenceType: apiObject
+    value: id.my-vpc.crossplane-system
+  route:
+    - cidrBlock: 0.0.0.0/0
+      gateway:
+        referenceType: apiObject
+        value: id.my-gateway.crossplane-system
+--
+apiVersion: network.aws.crossplane.io/v1alpha1
+kind: AwsRouteTableAssociation
+metadata:
+  namespace: crossplane-system
+  name: my-rt-association
+spec:
+  subnetId:
+    referenceType: apiObject
+    value: id.my-subnet-1.crossplane-system
+  routeTableId:
+    referenceType: apiObject
+    value: id.my-rt.crossplane-system
+--
+apiVersion: network.aws.crossplane.io/v1alpha1
+kind: AwsIamRole
+metadata:
+  namespace: crossplane-system
+  name: my-cluster-role
+spec:
+  nameFormat: clusterRole
+  assumeRolePolicy: policyJson
+--
+apiVersion: network.aws.crossplane.io/v1alpha1
+kind: AwsIamRolePolicyAttachment
+metadata:
+  namespace: crossplane-system
+  name: my-cluster-role-policy-attachment-1
+spec:
+  policy_arn: arn:aws:iam::aws:policy/EKSClusterPolicy
+  role:
+    referenceType: apiObject
+    value: name.my-cluster-role.crossplane-system
+--
+apiVersion: network.aws.crossplane.io/v1alpha1
+kind: AwsIamRolePolicyAttachment
+metadata:
+  namespace: crossplane-system
+  name: my-cluster-role-policy-attachment-2
+spec:
+  policy_arn: arn:aws:iam::aws:policy/AmazonEKSClusterPolicy
+  role:
+    referenceType: apiObject
+    value: name.my-cluster-role.crossplane-system
+--
+apiVersion: network.aws.crossplane.io/v1alpha1
+kind: AwsIamRolePolicyAttachment
+metadata:
+  namespace: crossplane-system
+  name: my-cluster-role-policy-attachment-2
+spec:
+  policy_arn: arn:aws:iam::aws:policy/AmazonEKSClusterPolicy
+  role:
+    referenceType: apiObject
+    value: name.my-cluster-role.crossplane-system
+
+### RDS connectivity related managed resources
+--
+apiVersion: network.aws.crossplane.io/v1alpha1
+kind: AwsDbSubnetGroup
+metadata:
+  namespace: crossplane-system
+  name: my-db-subnet-group
+spec:
+  nameFormat: subnetGroup
+  subnetIds:
+    - referenceType: apiObject
+      value: id.my-subnet-1.crossplane-system
+--
+apiVersion: network.aws.crossplane.io/v1alpha1
+kind: AwsSecurityGroup
+metadata:
+  namespace: crossplane-system
+  name: my-rds-sg
+spec:
+  nameFormat: rdsSg
+  vpcId:
+    referenceType: apiObject
+    value: id.my-vpc.crossplane-system
+  description: Cluster communication with worker nodes
+  ingress:
+    - fromPort: 3306
+      toPort: 3306
+      protocol: tcp
+      cidrBlocks:
+        - 0.0.0.0/0
+```
+
+Once all these connectivity managed resources are created, the resource classes for `EKSCluster` and `RDSInstance` can be configured as following:
+
+
+```yaml
+---
+apiVersion: core.crossplane.io/v1alpha1
+kind: ResourceClass
```

Comment:
These will need to be `RDSInstanceClass` and `EKSClusterClass`. I am planning on updating the existing GCP examples above to reflect this as well.